core/src/main/java/tech/pegasys/web3signer/core/multikey/metadata/SigningMetadataValidator.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.core.multikey.metadata;
+
+import tech.pegasys.web3signer.core.config.AzureAuthenticationMode;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+/** SigningMetadata custom validation which may not be performed by parsing libraries */
+public class SigningMetadataValidator {
+
+  public static void validate(final SigningMetadata metadata) throws SigningMetadataException {
```

Comment:
don't really like this class :/ The public "validate" function throws away information, which has to be re-created using the "instanceof" - and I'm wondering why we only validate 1 config type, and no others?

Reply:
let me try to use jackson annotations to perform the validation ... that will get rid of this class.